File: walkman-espresso/src/main/java/walkman/Walkman.java
package walkman;

import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import static java.lang.annotation.ElementType.METHOD;
import static java.lang.annotation.ElementType.TYPE;
import static java.lang.annotation.RetentionPolicy.RUNTIME;

@Retention(RUNTIME)
@Target({METHOD, TYPE})
public @interface Walkman {
  String tape() default "";
  TapeMode mode() default TapeMode.UNDEFINED;
  MatchRules[] match() default {};
}

File: walkman-junit/src/main/kotlin/walkman/Walkman.kt
package walkman

@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FUNCTION,
    AnnotationTarget.PROPERTY_GETTER,
    AnnotationTarget.PROPERTY_SETTER,
    AnnotationTarget.CLASS,
    AnnotationTarget.FILE)
annotation class Walkman(
    val tape: String = "",
    val mode: TapeMode = TapeMode.UNDEFINED,
    val match: Array<MatchRules> = arrayOf())
